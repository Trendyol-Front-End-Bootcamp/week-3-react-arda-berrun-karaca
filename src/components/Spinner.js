import React from 'react';
import './styles/Spinner.css';

function Spinner() {
  return (
    <img
      className="spinner"
      alt="Loading"
      src={`${process.env.PUBLIC_URL}/assets/loading-spinner.png`}
    />
  );
}

export default Spinner;
