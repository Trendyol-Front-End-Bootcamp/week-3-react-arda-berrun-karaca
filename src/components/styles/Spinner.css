.spinner {
    width: 50px;
    height: 50px;
    animation: rotating 3s linear infinite;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%,-50%);
}

@keyframes rotating {
    from {
      transform: rotate(0deg)
    }
    to {
      transform: rotate(360deg) 
    }
}