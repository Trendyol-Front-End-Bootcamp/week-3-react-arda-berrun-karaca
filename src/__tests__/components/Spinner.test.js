import React from 'react';
import renderer from 'react-test-renderer';

import Spinner from '../../components/Spinner';

describe('Spinner tests', () => {
  it('should be defined', () => {
    expect(Spinner).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = renderer.create(<Spinner />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
