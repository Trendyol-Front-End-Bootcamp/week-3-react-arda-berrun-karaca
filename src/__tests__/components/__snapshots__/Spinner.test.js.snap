// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests should render correctly 1`] = `
<img
  alt="Loading"
  className="spinner"
  src="/assets/loading-spinner.png"
/>
`;
